5.7.37 codership mysql wsrep 5.7.36 25.28

nitendra-bhosle opened the pull request.

temeo committed:
Merge tag 'mysql-5.7.36' into 5.7-v25-5.7.36

temeo committed:
Define wsrep version numbers in WSREP_VERSION file

bkandasa committed:
Update License Book

Approved-by: Bjorn Munch <bjorn.munch@oracle.com>

temeo committed:
Additional suppression for auth_sec.keyring_file_data_qa

When running the test in container environment, the lseek() call
to directory may also return EINVAL, for which there is no
suppression in the test. Added the suppression for the test
to work in all the environments where the test is run.

Commit:
Fix test galera.MW-44 to make it work with --ps-protocol

Commit:
Skip test galera.MW-328C under --ps-protocol

This test relies on wsrep_retry_autocommit, which has no effect under
ps-protocol.

Commit:
Fix galera.galera_flush under --ps-protocol

Do not enter TO ISOLATION on COM_STMT_PREPARE commands

Commit:
Fix test galera.galera_ftwrl_drain under --ps-protocol

Use an existing table for INSERT in the test, otherwise under
--ps-protocol, the INSERT fails due to the table not existing, instead
of the expected lock wait timeout.

Commit:
Skip test galera.galera_var_retry_autocommit under --ps-protocol

This test relies on wsrep_retry_autocommit, which has no effect under
ps-protocol.

Commit:
Disable tests MW-328[A|B] under --ps-protocol

See codership/mysql-wsrep-bugs#1130

abychko committed:
debian packages for mysql-wsrep-5.7 are not installable

[*] changed dependencies for mysql-wsrep-test-5.7 to add dependencies with
5.7 suffix
[*] added libxml-simple-perl dependency to test package

Commit:
Add sync waiting before statement prepare

When a statement is prepared, tables used in the statement may be
opened and checked for existence. Because of that, some tests (for
example galera_create_table_as_select) that CREATE a table in one node
and then SELECT from the same table in another node may result in errors
due to non existing table.
To make tests behave similarly under normal and PS protocol, we add a
call to sync wait before preparing statements that would sync wait
during normal execution.

sjaakola committed:
MDEV-26803 PA unsafety with FK cascade delete operation

This commit has a mtr test where two two transactions delete a row from
two separate tables, which will cascade a FK delete for the same row in
a third table. Second replica node is configured with 2 applier threads,
and the test will fail if these two transactions are applied in parallel.

The actual fix, in this commit, is to mark a transaction as unsafe for parallel applying
when it traverses into cascade delete operation.

sjaakola committed:
porting MDEV-27276 fix from mariadb

Here CREATE TABLE with foreign key contraints, will
get referenced tables appended as keys in the write set

sjaakola committed:
protection for CREATE TABLE ..LIKE

Added test scenario, where CREATE TABLE...LIKE.. creates a table with
FK constraints. Marking such replication as unsafe for PA.

temeo committed:
Add lsof into debian build dependencies

temeo committed:
Fix skip comment in galera_have_debug_sync.inc

The comment should not be wrapped in double quotes, as it
results in duplicated double quotes in XML report.

temeo committed:
Make galera_lock_table more deterministic

- Before moving on to execute actual test, verify that the
  tables have actually been created on node_2.
- Wait until applier has hit table metadata lock wait before
  verifying that no rows have been applied.
- Use read uncommitted transaction isolation to verify that
  tables are empty.

temeo committed:
Make galera_flush more deterministic

Ensure that insertion of new user into mysql.user gets
committed on node_2 before connecting with that user credentials.

temeo committed:
Increase timeout in galera_3nodes.galera_ipv6_xtrabackup-v2

temeo committed:
Merge man pages from 5.7.36

temeo committed:
Bump version to 5.7.36-25.28

Commit:
PS-8115 - Fix `CREATE EVENT DEFINER`

*Problem*:

When the user contains the character `@` this is associated to the
hostname.

*Solution*:

Do as 8.0 and consider that the hostname is the part from the last '@'
till the end of the value of `DEFINER`.

'@' is not allowed in the hostname.

Commit:
Implemented PS-8048 (Merge MySQL 5.7.37) (no conflicts)

https://jira.percona.com/browse/PS-8048

Merge tag 'mysql-5.7.37' into merge-mysql-5.7.37-part2

ldonoso committed:
Merge pull request #4681 from ldonoso/PS-8115-5.7

PS-8115 - Fix `CREATE EVENT DEFINER`

nitendra-bhosle committed:
Merge pull request #4625 from nitendra-bhosle/merge-mysql-5.7.37-part2

Merge mysql 5.7.37 part2

EvgeniyPatlan committed:
Update percona-server-5.7_builder.sh

EvgeniyPatlan committed:
Merge pull request #4687 from EvgeniyPatlan/release-5.7.37-40

Update percona-server-5.7_builder.sh

EvgeniyPatlan committed:
Update percona-server-5.7_builder.sh

EvgeniyPatlan committed:
Merge pull request #4688 from EvgeniyPatlan/release-5.7.37-40

Update percona-server-5.7_builder.sh

EvgeniyPatlan committed:
PS-7 fix build on cent8

EvgeniyPatlan committed:
Merge pull request #4689 from EvgeniyPatlan/release-5.7.37-40

PS-7 fix build on cent8

EvgeniyPatlan committed:
PS-7 fix dependencies

EvgeniyPatlan committed:
Merge pull request #4690 from EvgeniyPatlan/release-5.7.37-40

PS-7 fix dependencies

EvgeniyPatlan committed:
PS-7 fix version

EvgeniyPatlan committed:
Merge pull request #4691 from EvgeniyPatlan/release-5.7.37-40

PS-7 fix version

Commit:
PXC-3908: Merge PS 5.7.37 Part 2 (Resolving conflicts)

https://jira.percona.com/browse/PXC-3908

Note :  No conflict found during git merge

Merge tag 'Percona-Server-5.7.37-40' into 5.7.37-psmerge-part-2

Percona Server release 5.7.37-40

Commit:
PXC-3908: Merge codership mysql-wsrep-5.7.36-25.28 Part 1 (with conflicts)

https://jira.percona.com/browse/PXC-3908

Files having merge conflicts:
cmake/wsrep.cmake
mysql-test/suite/auth_sec/r/keyring_file_data_qa.result
mysql-test/suite/auth_sec/t/keyring_file_data_qa.test
mysql-test/suite/galera/r/MW-44.result
mysql-test/suite/galera/t/MW-44.test
mysql-test/suite/galera/t/galera_flush.test
mysql-test/suite/galera/t/galera_lock_table.test
packaging/deb-in/control.in
sql/sql_base.cc
sql/sql_parse.cc
storage/innobase/include/ha_prototypes.h
storage/innobase/row/row0ins.cc

Merge tag 'wsrep_5.7.36-25.28' into 5.7.37-codership-mysql-wsrep-5.7.36-25.28

Commit:
PXC-3908: Merge codership/wsrep_5.7.36-25.28 -> percona/pxc-5.7.37 (conflicts resolved)

https://jira.percona.com/browse/PXC-3908

Merge conflicts resolved

***
Updated WSREP_PATCH_VERSION to 57 in the wsrep.cmake file.

***
Updated WSREP-REVISION to reflect the upstream refresh revision
commit id: 02aadcc

venkatesh-prasad-v commented on WSREP_VERSION:
```diff
@@ -0,0 +1,2 @@
+WSREP_VERSION_API=25
+WSREP_VERSION_PATCH=28
```
In `cmake/wsrep.cmake`, we have set `WSREP_PATCH_VERSION=57`, but it is 28 here. This causes inconsistency in versioning. Could you please fix it?

venkatesh-prasad-v commented on cmake/wsrep.cmake:
```diff
@@ -16,29 +16,12 @@
-SET(WSREP_API_VERSION ${CMAKE_MATCH_1})
-STRING(REGEX MATCH "WSREP_VERSION_PATCH=([0-9]*)" _ ${wVER})
-SET(WSREP_PATCH_VERSION ${CMAKE_MATCH_1})
->>>>>>> wsrep_5.7.36-25.28
```
I think it is better we retain upstream's logic of fetching the `WSREP_PATCH_VERSION` from the WSREP_VERSION file. Could you please revert these changes?

it-percona-cla commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/not_signed)](https://cla-assistant.percona.com:443/percona/percona-xtradb-cluster?pullRequest=1621) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/percona-xtradb-cluster?pullRequest=1621) before we can accept your contribution.<br/>**2** out of **9** committers have signed the CLA.<br/><br/>:white_check_mark: nitendra-bhosle<br/>:white_check_mark: EvgeniyPatlan<br/>:x: sciascid<br/>:x: temeo<br/>:x: Luis Donoso<br/>:x: bkandasa<br/>:x: abychko<br/>:x: sjaakola<br/>:x: ldonoso<br/><hr/>**Luis Donoso** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/percona-xtradb-cluster?pullRequest=1621) it.</sub>

venkatesh-prasad-v reviewed (approved):
lgtm

Commit:
PXC-3908: Merge codership/wsrep_5.7.35-25.28 -> percona/pxc-5.7.37

https://jira.percona.com/browse/PXC-3908

Updated submodule (percona-xtradb-cluster-galera) pointer to 3.x-5.7

nitendra-bhosle commented:
Jenkins run : https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-param/293/
